### PR TITLE
Fix basePicker createGenericItem handling.

### LINKED ISSUE #3509: BasePicker createGenericItem creates nested structure
### Bug Report

- __Package version(s)__: 5.29.0

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: High

#### Describe the issue:

Using the createGenericItem on the TagPicker, the typing calls for it to return a ISuggestionModel<T>, doing so results in the new "tag"/item to  have the structure of a ```ISuggestionModel<T>``` instead of ```T```

The issue stems from BasePicker._onValidateInput() calls this.props.createGenericItem, which is expected to return a ```ISuggestionModel<T>``` (but it has been typed as any... )
It then proceeds to call suggestionStore.createGenericSuggestion with the item returned from the callback.
In the Suggestions controller, the item passed is "converted" into a ```ISuggestionModel<T>```

making the structrue as such:
```code
ISuggestionModel<T> {
   item: ISuggestinModel<T> {
      item: T;
      selected: boolean;
   }
}
```

#### Actual behavior:

creating a generic item, adds a ```ISuggestionModel<ISuggestionModel<T>>``` instead of an ```item<T>``` in the items list of the picker.

#### Expected behavior:

creating a generic item, adds a an ```item<T>``` in the items list of the picker.

### PR DESCRIPTION
**Pull request by ohritz**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #3509
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Added logic in the ```SuggestionsController.convertSuggestionsToSuggestionItems()``` to handle being called with either ```ISuggestionModel<T>``` or ```T```

It addresses an issue where when using a pickers createGenericItem, items are not resolved correctly.

### CODE REVIEW

**Comment by ohritz** [comment]
Please wait with merging.. 
had forgot to run the tests.. 
Will be pushing again once I've resolved some errors, 
and fixed formatting errors, that obfuscate the actual changes.

**Comment by ohritz** [comment]
the tests are passing now.
and I've tried to minimize the changes in formatting, let me know if I should do more.. 
Seems like It formats differently on save compared to format document in vs.code.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 1]
```diff
@@ -631,7 +631,7 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
 
   private _onValidateInput() {
     if (this.props.onValidateInput && (this.props.onValidateInput as any)(this.input.value) !== ValidationState.invalid && this.props.createGenericItem) {
-      let itemToConvert = (this.props.createGenericItem as any)(this.input.value, (this.props.onValidateInput as any)(this.input.value));
+      let itemToConvert = (this.props.createGenericItem as (input: string, ValidationState: ValidationState) => ISuggestionModel<T>)(this.input.value, (this.props.onValidateInput as any)(this.input.value));
```
[the remark this answers is not in the document]
Comment (ohritz) [reply packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: done !

File: packages/office-ui-fabric-react/src/components/pickers/Suggestions/SuggestionsController.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ export class SuggestionsController<T> {
   constructor() {
     this.suggestions = [];
     this.currentIndex = -1;
+    this._ensureSuggestionModel = this._ensureSuggestionModel.bind(this);
+    this._isSuggestionModel = this._isSuggestionModel.bind(this);
```
[the remark this answers is not in the document]
Comment (ohritz) [reply packages/office-ui-fabric-react/src/components/pickers/Suggestions/SuggestionsController.ts]: and done! :)

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 3]
```diff
@@ -631,7 +631,7 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
 
   private _onValidateInput() {
     if (this.props.onValidateInput && (this.props.onValidateInput as any)(this.input.value) !== ValidationState.invalid && this.props.createGenericItem) {
-      let itemToConvert = (this.props.createGenericItem as any)(this.input.value, (this.props.onValidateInput as any)(this.input.value));
+      let itemToConvert = this.props.createGenericItem(this.input.value, (this.props.onValidateInput)(this.input.value));
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Don't need these extra parens around `this.props.onValidateInput`

**Pull request merged**